crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -677,6 +691,9 @@ impl DerefMut for TimelineInnerStateTransaction<'_> {
 
 #[derive(Debug)]
 pub(in crate::timeline) struct TimelineInnerMetadata {
+    /// List of all the events as received in the timeline, even the ones that
+    /// are discarded in the timeline items.
+    all_events: VecDeque<EventMeta>,
```

Comment:
I guess because we need efficient front-insertion, we can't use `IndexMap`? That's pretty unfortunate, but I guess it should still perform well enough for a couple hundret or thousand events.

Reply:
Indeed